flycheck.el

```diff
@@ -4862,9 +4866,13 @@ In the latter case, show messages in the buffer denoted by
 variable `flycheck-error-message-buffer'."
   (when (and errors (flycheck-may-use-echo-area-p))
     (let ((messages (seq-map #'flycheck-error-format-message-and-id errors)))
-      (display-message-or-buffer (string-join messages "\n\n")
-                                 flycheck-error-message-buffer
-                                 'not-this-window))))
+      (let ((result (display-message-or-buffer (string-join messages "\n\n")
+                                               flycheck-error-message-buffer
+                                               'not-this-window)))
+        (if (window-live-p result)
+            (with-current-buffer (window-buffer result)
+              (when (not (eq major-mode 'flycheck-error-message-mode))
```

Comment:
This could be an `unless`, and I think the convention is to use `derived-mode-p`